src/core/include/openvino/op/util/roi_align_base.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+
+///
+/// \brief Base class for ROIAlignXXX operators.
+///
+class OPENVINO_API ROIAlignBase : public Op {
+public:
+    OPENVINO_OP("ROIAlignBase", "util");
+
+protected:
+    ROIAlignBase() = default;
+    /// \brief Constructs a ROIAlignBase operation.
+    ///
+    /// \param input           Input feature map {N, C, H, W}
+    /// \param rois            Regions of interest to pool over
+    /// \param batch_indices   Indices of images in the batch matching
+    ///                        the number or ROIs
+    /// \param pooled_h        Height of the ROI output features
+    /// \param pooled_w        Width of the ROI output features
+    /// \param sampling_ratio  Number of sampling points used to compute
+    ///                        an output element
+    /// \param spatial_scale   Spatial scale factor used to translate ROI coordinates
+    ROIAlignBase(const Output<Node>& input,
+                 const Output<Node>& rois,
+                 const Output<Node>& batch_indices,
+                 const int pooled_h,
+                 const int pooled_w,
+                 const int sampling_ratio,
+                 const float spatial_scale);
+
+public:
+    // Return size of second dimension of the ROI input.
+    // Needed for validation check.
+    virtual int get_rois_input_second_dim_size() const = 0;
+
+    void validate_and_infer_types() override;
+    bool visit_attributes(AttributeVisitor& visitor) override;
+
+    int get_pooled_h() const {
+        return m_pooled_h;
+    }
+
+    void set_pooled_h(const int h) {
+        m_pooled_h = h;
+    }
+
+    int get_pooled_w() const {
```

Comment:
Sorry, but this is the convention. I have checked other op impl and everyone is doing it this way. I have also copied it from previous roi align impl. So I will stick to convention.

However, I don't like it as well - they should be however defined as inline in header file but outside the class def. But - as I said - I prefer to stick to the convention. If you want to change it, you can always make a new pr and fix all other ops impl - just to have single convention.

Reply:
This is not a convention. We can not refactor everything. Please stick to the @praasz advice.

Reply:
First of all, you guys commenting on the code which was not implemented by me. That code was just moved from file roi_align.hpp, as necessary step to introduce a common base for roi operators. It means that this code has passed the interview already. What is more, those functions where implemented 11 months ago by @praasz himself(it doesn't really matter, but it is worth to point it out here).

Now, if you guys want to introduce a new code convention every 11 month - please do it in separate PR and fix all the places. If you "can't refactor everything" - it is better to NOT refactor at all - otherwise you create a mess: 
- in this PR, because it is not about changing the code convention, is is about adding new operator def. Ideally, every PR should do only one logical thing. Changing code convention is completely orthogonal to adding new op def.
- in whole codebase, since nobody knows what is really the convention here. Every file might be different, just because somebody "had a time to fix this file or didn't have". 

Now, if you think that changing that is necessary, please follow up with a new pr that will fix the code to the single convention.